### PR TITLE
Document the Plant controller

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area documentation
/kind task

**What this PR does / why we need it**:
This part documents the `Plant` controller in GCM.

**Which issue(s) this PR fixes**:
Part of #3405

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/concepts/controller-manager.md [hunk 1]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
 
 Refer to [GEP-15](../proposals/15-manage-bastions-and-ssh-key-pair-rotation.md) for more information on the lifecycle of
 `Bastion` resources.
+
+### "Plant" Controller
+
+Using the `Plant` resource an external Kubernetes cluster (not managed by Gardener) can be registered to Gardener. Gardener Controller Manager is the component that is responsible for the `Plant` resource reconciliation. As part of the reconciliation loop the Gardener Controller Manager performs health checks on the the Plant and gathers more information about the underlying cluster itself - all of this information serve for monitoring purposes of the underlying Kubernetes cluster.
```
Comment (ialidzhikov) [inline docs/concepts/controller-manager.md]:
```suggestion
Using the `Plant` resource an external Kubernetes cluster (not managed by Gardener) can be registered to Gardener. Gardener Controller Manager is the component that is responsible for the `Plant` resource reconciliation. As part of the reconciliation loop the Gardener Controller Manager performs health checks on the Plant and gathers more information about the underlying cluster itself - all of this information serve for monitoring purposes of the external Kubernetes cluster.
```

File: docs/concepts/controller-manager.md [hunk 2]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
...
+
+`Plant` controller runs periodically and perform health checks that if, `API Server` and all the `Nodes` of that `Plant`'s Kubernetes cluster are up and running.
```
Comment (ialidzhikov) [inline docs/concepts/controller-manager.md]:
```suggestion
The `Plant` controller runs periodically and performs health checks for API Server availability and cluster `Node`s healthiness.
```

File: docs/concepts/controller-manager.md [hunk 3]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
...
+* `syncPeriod`: The duration how often the Plant resource is reconciled. The default value is `30s`.
+* `concurrentSyncs`: The amount of goroutines scheduled for reconciling events. The default value is `5`.
+
+The `Plant` resource reports following 5 fields of Kubernetes cluster:
+
+- CloudType
+- Region
+- K8sVersion
+- API Server
+- Nodes
```
Comment (ialidzhikov) [inline docs/concepts/controller-manager.md]:
```suggestion
The `Plant` resource reports the following information for the external Kubernetes cluster:
- Cluster information
  - Cloud provider information - the cloud provider type and region are maintained in the `Plant` status (`.status.clusterInfo.cloud`)
  - Kubernetes version - the Kubernetes version is maintained in the `Plant` status (`.status.clusterInfo.kubernetes.version`)
- Cluster status
  - API Server availability - maintained as condition with type `APIServerAvailable`
  - Cluster `Node`s healthiness - maintained as condition with type `EveryNodeReady`
```

**Review by ialidzhikov — COMMENTED** [review]
Minor suggestions, otherwise lgtm

**Review by ialidzhikov — DISMISSED** [review]
/lgtm

File: docs/concepts/controller-manager.md [hunk 4]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
 
 Refer to [GEP-15](../proposals/15-manage-bastions-and-ssh-key-pair-rotation.md) for more information on the lifecycle of
 `Bastion` resources.
+
+### "Plant" Controller
+
+Using the `Plant` resource an external Kubernetes cluster (not managed by Gardener) can be registered to Gardener. Gardener Controller Manager is the component that is responsible for the `Plant` resource reconciliation. As part of the reconciliation loop the Gardener Controller Manager performs health checks on the Plant and gathers more information about the underlying cluster itself - all of this information serve for monitoring purposes of the external Kubernetes cluster.
```
Comment (rfranzke) [inline docs/concepts/controller-manager.md]:
```suggestion
Using the `Plant` resource, an external Kubernetes cluster (not managed by Gardener) can be registered to Gardener. Gardener Controller Manager is the component that is responsible for the `Plant` resource reconciliation. As part of the reconciliation loop, the Gardener Controller Manager performs health checks on the external Kubernetes cluster and gathers more information about it - all of this information serve for monitoring purposes of the external Kubernetes cluster.
```

File: docs/concepts/controller-manager.md [hunk 5]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
...
+
+The `Plant` controller runs periodically and performs health checks for API Server availability and cluster `Node`s healthiness.
```
Comment (rfranzke) [inline docs/concepts/controller-manager.md]: This is a bit duplicative compared to the block above (you already mention health checks).

File: docs/concepts/controller-manager.md [hunk 6]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
...
+
+The component configuration of the Gardener Controller Manager offers to configure the following options for the plant controller:
+* `syncPeriod`: The duration how often the Plant resource is reconciled. The default value is `30s`.
```
Comment (rfranzke) [inline docs/concepts/controller-manager.md]:
```suggestion
* `syncPeriod`: The duration how often the Plant resource is reconciled, i.e., how often health checks are performed. The default value is `30s`.
```

File: docs/concepts/controller-manager.md [hunk 7]
```diff
@@ -187,3 +187,21 @@ deletion timestamp has been set already.
...
+* `concurrentSyncs`: The amount of goroutines scheduled for reconciling events. The default value is `5`.
```
Comment (rfranzke) [inline docs/concepts/controller-manager.md]:
```suggestion
* `concurrentSyncs`: The amount of goroutines scheduled for reconciling events, i.e., the number of possible parallel reconciliations. The default value is `5`.
```

**Comment by ialidzhikov** [comment]
Can you rebase? It seems that there is a failing build step because of https://github.com/gardener/gardener/pull/4757.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**